dotnet/src/SemanticKernel/Kernel.cs

```diff
@@ -200,15 +197,15 @@ public async Task<SKContext> RunAsync(ContextVariables variables, CancellationTo
                 if (context.ErrorOccurred)
                 {
                     this.Logger.LogError("Function call fail during pipeline step {0}: {1}.{2}. Error: {3}",
-                        pipelineStepCount, f.SkillName, f.Name, context.LastErrorDescription);
+                        pipelineStepCount, f.SkillName, f.Name, context.LastException?.Message);
                     return context;
                 }
             }
             catch (Exception e) when (!e.IsCriticalException())
             {
                 this.Logger.LogError(e, "Something went wrong in pipeline step {0}: {1}.{2}. Error: {3}",
                     pipelineStepCount, f.SkillName, f.Name, e.Message);
-                context.Fail(e.Message, e);
+                context.LastException = e;
```

Comment:
Do we want to re-throw it here?

Reply:
Same as below... that changes the flow from the caller perspective: currently, a failure in the SK function will allow the RunAsync() call to complete, then the caller can check the ErrorOccurred property. If we throw here, the RunAsync call itself will fail, which means that they won't get a chance to get to the ErrorOccurred property.

We may need to overhaul this with a breaking change. To coordinate with @SergeyMenshykh, and will address this if needed once his exception handling PR is complete.